azurerm/resource_arm_container_group.go

```diff
@@ -146,16 +147,48 @@ func resourceArmContainerGroup() *schema.Resource {
 						},
 
 						"port": {
-							Type:         schema.TypeInt,
-							Optional:     true,
-							ForceNew:     true,
-							ValidateFunc: validation.IntBetween(1, 65535),
+							Type:          schema.TypeInt,
+							Optional:      true,
+							ForceNew:      true,
+							ConflictsWith: []string{"container.0.ports"},
+							Deprecated:    "Use `ports` instead.",
+							ValidateFunc:  validate.PortNumber,
+						},
+
+						"ports": {
+							Type:          schema.TypeList,
+							Optional:      true,
+							ForceNew:      true,
+							ConflictsWith: []string{"container.0.port", "container.0.protocol"},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"protocol": {
+										Type:             schema.TypeString,
+										Optional:         true,
+										ForceNew:         true,
+										DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+										Default:          string("TCP"),
+										ValidateFunc: validation.StringInSlice([]string{
+											"tcp",
```

Comment:
Could we use the enum constants from the SDK `containerinstance.ContainerNetworkProtocol` here?